### DISCUSSION TITLE
Log drains

### DISCUSSION THREAD

**Discussion by eli5zen** [supabase/supabase#7572] [Feature Requests] [answered]
### Update - Log drains is now in private alpha, please see this discussion [here](https://github.com/orgs/supabase/discussions/26650).

---

Hello :)

## TL;DR

I think it would be valuable to have the ability to forward logs from supabase to an external logging provider that supplies a TLS-enabled token-based auth [syslog](https://datatracker.ietf.org/doc/html/rfc5424) endpoint (e.g., Papertrail, Datadog, newrelic).

### Better visibility for traceability

In a complex system with application services and a DB, many logs are generated. To efficiently debug a problem, logs generated during a request in the application code would reside in the same place together with the DB logs.

That would allow searching and filtering logs based on a context (e.g., user_id) to have a clear view of what happened during a request.

### Better visibility for security

Actions that are done by users within the superbase platform (e.g., changing a DB size) should be logged to allow generating an audit trail. Assuming other services (like hosting platforms of the deployed code) forward their audit trail to a centralized logging provider, the DB audit trail would ideally reside with them, to give the system operators clear visibility of what happened in the system and by who.

### (Maybe) lower cost on the supabase side to run a project

If an account chooses to forward all logs generated by supabase servers to an external logging provider, from supabase's perspective, a smaller amount of logs should be saved within the supabase platform (maybe only the data needed to support features like dashboard and analytics, if the account has opted-in to those features).

Would love to hear your thoughts on this :)
Thanks!

**Comment by mintavenger** [comment]
The existing log UI is very nice but limited noting the issues @thisis-ez raised above. It would be most excellent to have a way to pump these logs into a centralized location. For an example, you can look at the Datadog<>Vercel integration for log integration. As Supabase gets integrated into larger products observability will be a must. Thanks for your great developer experience, but also need to think about the devops experience once you get your product up and running.

**Reply by pax-k** [reply]
I agree, this is a must

**Comment by oscargws** [comment]
Following this, this is a massive use case for my org, we have our logs stored in Datadog and need to forward our supabase ones in.

**Reply by mewc** [reply]
agreed

**Comment by Ziinc** [comment]
Thanks for the feedback. There is currently ongoing work for improving the logging experience with increased Logflare integration into the stack.

For monitoring purposes, all projects have a prom-compatible endpoint exposed for [project infra metrics](https://supabase.com/docs/guides/platform/metrics).

**Comment by saravmajestic** [comment]
+1. We need to alert the teams based on errors/logs, and as we don't have any way of doing that in supabase, it will be a must to have 3rd party integration or log forwarding feature.

**Comment by cullophid** [comment]
+1 this would be amazing

**Comment by askkaz** [comment]
+1 datadog

**Comment by istx25** [comment]
+1 axiom

**Comment by maxdeichmann** [comment]
+1 betterstack

**Comment by erik-beus** [comment]
Honeycomb integration would be 🔥

**Comment by mosnicholas** [comment]
+1 for datadog

**Comment by eleleung** [comment]
+1 datadog please

**Comment by Ziinc** [comment, marked as answer]
Log drains is now in private alpha, please see this discussion [here](https://github.com/orgs/supabase/discussions/26650)
